src/Core/Auth/Identity/TokenProviders/DuoUniversalTokenService.cs

```diff
@@ -0,0 +1,177 @@
+﻿using Bit.Core.Auth.Models;
+using Bit.Core.Auth.Models.Business.Tokenables;
+using Bit.Core.Context;
+using Bit.Core.Entities;
+using Bit.Core.Settings;
+using Bit.Core.Tokens;
+using Duo = DuoUniversal;
+
+namespace Bit.Core.Auth.Identity.TokenProviders;
+
+/// <summary>
+/// OrganizationDuo and Duo TwoFactorProviderTypes both use the same flows so both of those Token Providers will
+/// have this class injected to utilize these methods
+/// </summary>
+public interface IDuoUniversalTokenService
```

Comment:
Non-blocking Suggestion: we should generally keep interfaces and implementations in separate files for separation of concerns.